### PR TITLE
Import DataSource - GPX/KML Files support (06)

### PR DESCRIPTION
**Pull request by adube**
This PR introduces the support of GPX and KML files into the ImportDataSource gmf component. Featuring:

 * a new `ngeo.datasource.File`, that receives the features, creates a vector layer
 * a new `ngeo.datasource.FileGroup` that handles the synchronization of the visibility properties from the file data sources and their layer (DataSource is master, Layer is slave)
 * addition/removal of the layer to the map by the external data sources manager
 * addition/removal to the layer tree

## Todo

 * [x] Wait for #2988 to be merged first, then rebase onto master
 * [x] Review

## How to test

Example files were added to the examples/data directory, in gmf.  You can test with those.

### CODE REVIEW

**Comment by adube** [comment]
Ready for review.

File: contribs/gmf/src/datasource/externaldatasourcesmanager.js [hunk 1]
```diff
@@ -433,6 +482,89 @@ gmf.datasource.ExternalDataSourcesManager = class {
...
+        fileGroup.addDataSource(dataSource);
+
+        // (4) Recenter the map view onto its extent
+        this.map_.getView().fit(dataSource.extent);
+
+        // (5) Finally, add it to the ngeo collection
+        this.ngeoDataSources_.push(dataSource);
+      },
+      (rejections) => {
+        // TODO, handle rejection...
```
Comment (fredj) [inline contribs/gmf/src/datasource/externaldatasourcesmanager.js]: please add a `goog.asserts.fail`

**Comment by adube** [comment]
Required corrections were applied.  Thanks for the review.

Will merge once Travis is green.

**Comment by adube** [comment]
Travis failed 2 times in a row because of issues unrelated to this PR.  I'll re-start it for a 3rd time.  Please, if it fails again, decide whether or not to merge it anyway.

**Comment by fredj** [comment]
I've also restarted travis, merge anyway if the error persist (ressource error on openstreetmap )

**Comment by fredj** [comment]
it's green, merging. Thanks @adube

**Pull request merged**